docs/integrations/github_actions/aws_oidc.md

```diff
@@ -0,0 +1,162 @@
+
+```yaml
+permissions:
+  id-token: write
+  contents: write
+```
+
+### Steps
+
+A workflow comprises one or more jobs that run in parallel, each with one or more steps that run in order. Our [example](https://github.com/turbot/steampipe-samples/blob/main/all/github-actions-oidc/aws/steampipe-sample-aws-workflow.yml#L14) defines a single job with a series of steps that authenticate to AWS, install Steampipe, run a compliance benchmark and save the results to the repository.
+
+Here's the step that configures the credentials Steampipe will use to access AWS.
+
+```yaml
+- name: "Configure AWS credentials"
+  id: config-aws-auth
+  uses: aws-actions/configure-aws-credentials@v1-node16
+  with:
+    role-to-assume: ${{ secrets.OIDC_AWS_ROLE_TO_ASSUME }}
+    role-session-name: "steampipe-demo"
+    role-duration-seconds: 900
+    aws-region: "us-east-1"
+```
+
+Once the cloud provider successfully validates the claims presented in the OIDC JWT ID token, it then provides a short-lived access token that is available only for the duration of the job. The short-lived access token is exported as environment variables like AWS_DEFAULT_REGION, AWS_REGION, AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY and AWS_SESSION_TOKEN.
+Steampipe will seamlessly load these short-lived [Credentials from Environment Variables](https://hub.steampipe.io/plugins/turbot/aws#credentials-from-environment-variables) to run the benchmark.
+
+Before running the compliance benchmark, create a new folder in your GitHub repository to save the benchmark outputs. Update the `export` argument in the below step. In our example, we will save the outputs to the folder `steampipe/benchmarks/aws`.
+
+Here's the step that runs the [AWS Compliance](https://hub.steampipe.io/mods/turbot/aws_compliance) mod.
+
+```yaml
+- name: "Run Steampipe benchmark"
+  id: steampipe-benchmark
+  run: |
+
+    # Install the Steampipe AWS Compliance mod
+    steampipe mod install github.com/turbot/steampipe-mod-aws-compliance 
+    cd .steampipe/mods/github.com/turbot/steampipe-mod-aws-compliance*
+    # Run the AWS CIS v1.5.0 benchmark
+    steampipe check benchmark.cis_v150 --export=${GITHUB_WORKSPACE}/steampipe/benchmarks/aws/cis_v150_"$(date +"%d_%B_%Y")".html --output=none
+```
+
+
+Here's the step that pushes the benchmark to your repository. Update the `working-directory` to the folder created in the above step. This should be the same location used in the above `export` argument.
+
+```yaml
+- name: "Commit the file to github"
+  id: push-to-gh
+  working-directory: steampipe/benchmarks/aws
+  run: |
+
+    git config user.name github-actions
+    git config user.email github-actions@github.com
+    git add cis_v150_"$(date +"%d_%B_%Y")".html 
+    git commit -m "Add Steampipe Benchmark Results"
+    git push
+```
+
+### Run the workflow
```

Comment:
### Run the workflow on-demand

Reply:
I'd also make this a level 2 heading and leave it at the bottom. So if we re-order the sections it would be:

Define the Workflow
Setup GitHub's OIDC provider for AWS
Run the workflow on-demand